app/controllers/application_controller.rb

```diff
@@ -1,4 +1,18 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   include SessionsHelper
+  before_action :not_logged_in?
+
+  private
+
+    def not_logged_in?
+      redirect_to login_path if !logged_in?
+    end
+
+    def correct_user?
+      @user = User.find(params[:id])
+      unless @user == current_user
+        redirect_to root_url
```

Comment:
What do you think about adding a flash alert to this redirect to let the user know why they're being rerouted?